packages/updating-element/src/updating-element.ts

```diff
@@ -788,17 +815,14 @@ export abstract class UpdatingElement extends HTMLElement {
       this._instanceProperties!.forEach((v, p) => ((this as any)[p] = v));
       this._instanceProperties = undefined;
     }
-    // create renderRoot before first update.
-    if (!this.hasUpdated) {
-      (this as {
-        renderRoot: Element | DocumentFragment;
-      }).renderRoot = this.createRenderRoot();
-    }
     let shouldUpdate = false;
     const changedProperties = this._changedProperties;
     try {
       shouldUpdate = this.shouldUpdate(changedProperties);
       if (shouldUpdate) {
+        this._controllers?.forEach((c) => c.willUpdate?.(changedProperties));
+        this.willUpdate(changedProperties);
```

Comment:
We need to add calling `willUpdate` during SSR, if that's the intention. Either add to this PR, or file an issue?

Reply:
Tracked via https://github.com/Polymer/lit-html/issues/1417